fix: pass args as array to Function.prototype.apply

DanielHeckrath opened the pull request:
Function.prototype.apply has call signature `apply(thisArg, argsArray)` so we cannot use spread operator on the arguments

DanielHeckrath committed:
fix: properly call original function in @Log decorator

jmcdo29 commented:
Ah! You are correct! I was thinking of `impl(...args).bind(this)`. Good catch. Could you make a changeset for this (patch should be good) that has a description of the change. Just run `pnpm changeset` and follow the wizard

DanielHeckrath commented:
@jmcdo29 I have updated the PR with a changeset. I have also changed the `target` variable in apply to `this` as that was causing additional problems when a second function that was decorator with @Log was called from another decorated function.

jmcdo29 commented:
> @jmcdo29 I have updated the PR with a changeset. I have also changed the `target` variable in apply to `this` as that was causing additional problems when a second function that was decorator with @log was called from another decorated function.

Awesome. I think I remember discussing this with someone on Discord as well, but hadn't made it around to updating the code. Thanks for making this. Let me see what's going on with CI and if it can be ignored or not
